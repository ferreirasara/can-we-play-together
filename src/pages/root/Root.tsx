import { Alert, Box, Link, Paper, Snackbar, Typography, useMediaQuery, useTheme } from "@mui/material";
import { useState } from "react";
import GameList from "../../components/GameList";
import SearchForm from "../../components/SearchForm";
import { GameDetails, getPlayerOwnedGames } from "../../services/player";

export default function Root() {
  const theme = useTheme();
  const isDesktop = useMediaQuery(theme.breakpoints.up('sm'));

  const [loading, setLoading] = useState<boolean>(false);
  const [games, setGames] = useState<GameDetails[]>();
  const [snackbarOpen, setSnackbarOpen] = useState<boolean>(false);
  const [snackbarMessage, setSnackbarMessage] = useState<string>('');

  const handleClick = async (userId1: string, userId2: string) => {
    setGames([]);
    setLoading(true);
    try {
      const games = await getPlayerOwnedGames(userId1 || '', userId2 || '');

      if (!games?.success) {
        setSnackbarOpen(true);
        setSnackbarMessage(games?.message || 'An error occurred.');
      } else {
        setGames(games?.gamesInCommon);
      }
    } catch (error: any) {
      console.log(error);
      setSnackbarOpen(true);
      setSnackbarMessage(error?.message);
    }
    setLoading(false);
  }

  return (
    <Box
      height='100vh'
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
    >
      <Paper
        elevation={0}
        sx={{
          borderRadius: '16px',
          maxWidth: 700,
          maxHeight: isDesktop ? 'auto' : '85%',
          display: 'flex',
          flexDirection: 'column',
          my: isDesktop ? 12 : 0,
          p: 2,
        }}
      >
        <SearchForm
          handleClick={handleClick}
          loading={loading}
        />
        {games?.length ? <GameList games={games} /> : null}
      </Paper>
      <Box
        position="absolute"
        bottom={5}
      >
        <Typography color="#FFF">
          Developed by <Link underline="none" target="_blank" rel="noopener" href="https://github.com/ferreirasara">Sara Ferreira</Link>
        </Typography>
      </Box>
      <Snackbar
        anchorOrigin={{ horizontal: 'center', vertical: 'top' }}
        open={snackbarOpen}
        autoHideDuration={6000}
        onClose={() => setSnackbarOpen(false)}
      >
        <Alert
          onClose={() => setSnackbarOpen(false)}
          severity="error"
          variant="filled"
          sx={{ maxWidth: '100%' }}
        >
          <Typography>{snackbarMessage}</Typography>
        </Alert>
      </Snackbar>
    </Box >
  );
}
