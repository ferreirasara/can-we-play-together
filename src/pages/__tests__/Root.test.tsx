import renderer from 'react-test-renderer';
import Root from '../root/Root';

it('renders correctly', () => {
  const filters = renderer
    .create(<Root />)
    .toJSON();
  expect(filters).toMatchSnapshot();
});