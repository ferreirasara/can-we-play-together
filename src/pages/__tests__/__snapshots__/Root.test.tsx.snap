// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-15ue26l"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-n3qv3e-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-ewzdx8"
    >
      <div
        className="MuiBox-root css-1embff6"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="user1"
            id="user1-label"
          >
            Your username
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="user1"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Your username
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="user2"
            id="user2-label"
          >
            Your friend's username
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="user2"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Your friend's username
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1oob06v-MuiTypography-root"
        onClick={[Function]}
      >
        How can I find my steamid?
      </span>
      <button
        className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-sppx0e-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        disabled={false}
        id=":r3:"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Can we play together?
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-uxlygp"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1r998hm-MuiTypography-root"
    >
      Developed by 
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1cmksfb-MuiTypography-root-MuiLink-root"
        href="https://github.com/ferreirasara"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        rel="noopener"
        target="_blank"
      >
        Sara Ferreira
      </a>
       
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InfoOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </button>
    </p>
  </div>
</div>
`;
